src/libraries/System.Text.Json/System.Text.Json.sln

```diff
@@ -21,9 +21,9 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "System.Text.Encodings.Web",
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "System.Text.Encodings.Web", "..\System.Text.Encodings.Web\src\System.Text.Encodings.Web.csproj", "{9BCCDA15-8907-4AE3-8871-2F17775DDE4C}"
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "System.Text.Json.SourceGeneration", "gen\System.Text.Json.SourceGeneration.Roslyn3.11.csproj", "{04AEB008-EE4F-44DE-A361-2DBD2D0FD6A4}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "System.Text.Json.SourceGeneration.Roslyn3.11", "gen\System.Text.Json.SourceGeneration.Roslyn3.11.csproj", "{04AEB008-EE4F-44DE-A361-2DBD2D0FD6A4}"
```

Comment:
@eerhardt FYI I changed this while using Codespaces because apparently the sdk on linux will complain about the ambiguity:
```
/home/eirik/devel/dotnet/runtime/src/libraries/System.Text.Json/System.Text.Json.sln : Solution file error MSB5004: The solution file has two projects named "System.Text.Json.SourceGeneration".
```

Reply:
See https://github.com/dotnet/runtime/pull/60871#discussion_r738601748. The fix is coming with the next update to slngen.

Reply:
Should I revert the change? The sln project name doesn't have any bearing on the build artifacts.

Reply:
No. I wouldn’t revert the change. I’ve made similar changes in other solutions to help with my productivity. When we take an update to slngen, it will be fixed.